src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -73,6 +96,24 @@ public void onClean(GitSCM scm, GitClient git) throws IOException, InterruptedEx
     }
 
     @Override
+    public GitClient decorate(GitSCM scm, GitClient git) throws IOException, InterruptedException, GitException {
+        for (Credentials creds : credentials) {
+            StandardCredentials credentials = CredentialsMatchers
+                .firstOrNull(CredentialsProvider.lookupCredentials(StandardCredentials.class,
+                            (Item) null,
+                            ACL.SYSTEM,
```

Comment:
You want to be using the user that the build is running as not `ACL.SYSTEM` otherwise you cannot inject the user's own credentials via a credentials parameter or work correctly with the Authorize Project Plugin